Nos/Views/Discover/DiscoverGrid.swift

```diff
@@ -0,0 +1,135 @@
+                                scrollingPicker
+
+                                ForEach(authors) { author in
+                                    AuthorCard(author: author) {
+                                        router.push(author)
+                                    }
+                                    .padding(.horizontal, 13)
+                                    .padding(.top, 5)
+                                    .readabilityPadding()
+                                    .task {
+                                        // TODO: optimize. Probably only needed once per author.
+                                        subscriptions[author.id] =
+                                            await relayService.requestMetadata(
+                                                for: author.hexadecimalPublicKey,
+                                                since: author.lastUpdatedMetadata
+                                            )
+                                    }
+                                }
+                            }
+                            .padding(.bottom, 16)
+                        }
+                        .doubleTapToPop(tab: .discover) { proxy in
+                            if let firstAuthor = authors.first {
+                                proxy.scrollTo(firstAuthor.id)
+                            }
+                        }
+                    case .empty:
+                        EmptyView()
+                    case .loading, .stillLoading:
+                        FullscreenProgressView(
+                            isPresented: .constant(true),
+                            text: searchController.state == .stillLoading ?
+                            String(localized: .localizable.notFindingResults) : nil
+                        )
+                    case .results:
+                        ScrollView {
+                            LazyVStack {
+                                ForEach(searchController.authorResults) { author in
+                                    AuthorCard(author: author) {
+                                        router.push(author)
+                                    }
+                                    .padding(.horizontal, 15)
+                                    .padding(.top, 10)
+                                    .readabilityPadding()
+                                }
+                            }
+                            .padding(.top, 5)
+                        }
+                        .doubleTapToPop(tab: .discover) { proxy in
+                            if let firstAuthor = searchController.authorResults.first {
+                                proxy.scrollTo(firstAuthor.id)
+                            }
+                        }
+                    }
+                }
+                .preference(key: SizePreferenceKey.self, value: geometry.size)
+            }
+        }
+    }
+
+    var scrollingPicker: some View {
```

Comment:
I didn't understand what this was without reading the code. Maybe a less generic name like `categoryBar` or something is better?